### PR TITLE
Fixing up & adding to PR template

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
## Description

Cleanup of PR template and added checklists for PR creator intent
Issue: https://app.zenhub.com/workspaces/ua-ag-data-science-5a57a3198339f11ba1c85775/issues/agpipeline/issues-and-projects/162

## Types of changes

Updated template to remove unused [front matter](https://help.github.com/en/github/working-with-github-pages/about-github-pages-and-jekyll#front-matter)
Added some checklists to help clarify aspects of PRs (timeliness and main intent)

## Checklist:

- [x] I have read the documentation on [CONTRIBUTING](https://github.com/AgPipeline/Organization-info) 
- [ ] My updates require a change to the documentation 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

## Focus of Pull Request:
All comments are welcome regardless of which items are checked off.
- [ ] All of the below (no need to check off the other boxes)
- [ ] There are no errors or inconsistencies
- [x] Everything appears clear and easy to understand
- [x] Suggest improvements
- [ ] Identify areas where functionality and/or documentation is missing

## Time frame for completing Pull Request review (from creation date):
- [ ] As soon as possible
- [x] In the next two weeks
- [ ] In the next month

## Hints to reviewers:

N/A

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -38,6 +30,20 @@ eg: Bug fix for widget so that it now correctly thrombs the tubas to provide cle
...
 - [ ] All new and existing tests passed
 
+## Time frame for completing Pull Request review (from creation date):
+- [ ] As soon as possible
+- [ ] In the next two weeks
+- [ ] In the next month
+
+## Focus of Pull Request:
+<!--- Indicate the main focus of the PR. Additional information can be added below in the Hints section -->
+All comments are welcome regardless of which items are checked off.
```
Comment (dlebauer) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Reviewers should focus on the following aspects of this pull request. All comments are welcome regardless of which items are checked off.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -38,6 +30,20 @@ eg: Bug fix for widget so that it now correctly thrombs the tubas to provide cle
...
+- [ ] All of the below (no need to check off the other boxes)
+- [ ] There are no errors or inconsistencies
+- [ ] Everything appears clear and easy to understand
+- [ ] Suggest improvements
+- [ ] Identify areas where functionality and/or documentation is missing
+
```
Comment (dlebauer) [inline .github/PULL_REQUEST_TEMPLATE.md:46]:
```suggestion
- [ ] Specific aspects to focus on are listed below
- [ ] Specific aspects and known issues _not to_ focus on are listed below
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -38,6 +30,20 @@ eg: Bug fix for widget so that it now correctly thrombs the tubas to provide cle
...
 
+## Time frame for completing Pull Request review (from creation date):
+- [ ] As soon as possible
+- [ ] In the next two weeks
+- [ ] In the next month
+
+## Focus of Pull Request:
+<!--- Indicate the main focus of the PR. Additional information can be added below in the Hints section -->
+All comments are welcome regardless of which items are checked off.
+- [ ] All of the below (no need to check off the other boxes)
```
Comment (dlebauer) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] Inform others of new features and changes
```

I don't the 'all of the below' is necessary; if it is, an 'all of the above' is a more familiar pattern

**Review by dlebauer — COMMENTED** [review]
This is a good chance to figure out what belongs on OSF, what on CONTRIBUTING.md, and what in the PR Template itself (or perhaps this is out of scope here? the whole point is to make PRs more focused ... 

Perhaps also update the PR section of CONTRIBUTING.md and / or  our generic PR docs on the [osf wiki](https://osf.io/tzmhp/wiki/Using%20Git%20and%20GitHub/)  with guidelines for writing an effective PR and writing an effective PR review, including something like: 
  - guide from the github blog https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/
  - suggestions for compassionate code reviews https://github.com/margarethchan/Compassionate-Code-Reviews#compassionate-code-reviews

Specific suggestions:

* can state that style review is not necessary if there is a linter
* can state that it is not necessary to run tests if there is automated testing 
* instruct PR authors to request specific people to look at particular aspects of the code
* link to CONTRIBUTING.md and our OSF page?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (Chris-Schnaufer) [inline .github/PULL_REQUEST_TEMPLATE.md:46]: I'm not sure we need checkboxes for these if they exist in the Hints section below. Perhaps combine them into "Important information in Hints section" or similar? The hints themselves should say they they are specific somethings to focus on or not IMO

**Comment by Chris-Schnaufer** [comment]
@dlebauer Some q's about what you wrote:
> Specific suggestions:

Are these to be incorporated into the PR template or in the other documentation. I think having another checkbox under "Checklist" that's similar to "Have read how to make a PR" with a link to the documentation would keep the already cluttered PR template cleaner

> can state that it is not necessary to run tests if there is automated testing

Can you clarify if this is so that people don't try to run tests locally when there's automated testing?
If this is in reference to "I have added tests to cover my changes" and "All new and existing tests passed" checkboxes, the purpose of those lines is to indicate that all new code (and existing code, but we're not there yet) should have tests (unit, integration, etc).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -38,6 +30,20 @@ eg: Bug fix for widget so that it now correctly thrombs the tubas to provide cle
...
+Reviewers should focus on the following aspects of this pull request. All comments are welcome regardless of which items are checked off.
+- [ ] Inform others of new features and changes
+- [ ] There are no errors or inconsistencies
+- [ ] Everything appears clear and easy to understand
+- [ ] Suggest improvements
+- [ ] Identify areas where functionality and/or documentation is missing
+
 ## Hints to reviewers:
 <!--- Provide additional information to help reviewers focus on meaningful changes -->
 <!--- For example: Corrected indentation in function foo99() - no text or code was changed -->
```
Comment (dlebauer) [inline .github/PULL_REQUEST_TEMPLATE.md:49]:
```suggestion
<!--- For example: Corrected indentation in function foo99() - no text or code was changed -->

<!-- **specific aspects to focus on or ignore**-->
<!-- you can @mention specific people to have them focus on specific changes -->
<!-- ask reviewers to ignore any known issues or future features that are out of scope for this PR-->
```

**Comment by Chris-Schnaufer** [comment]
Added issue for making additional documentation changes: https://github.com/AgPipeline/issues-and-projects/issues/186

**Comment by Chris-Schnaufer** [comment]
Merging so we can use this template while outstanding issues are resolved.

**Pull request merged**